diesel/src/query_builder/ast_pass.rs

```diff
@@ -227,7 +227,20 @@ where
         Ok(())
     }
 
-    pub(crate) fn push_bind_param_value_only<T, U>(&mut self, bind: &'b U) -> QueryResult<()>
+    /// Push a value onto the given query to be sent separate from the SQL
+    ///
+    /// This method affects multiple AST passes. It should be called at the
+    /// point in the raw SQL is inserted. This assumes the parameter placeholder
+    /// (`$1` on PG, `?` on other backends) is already inserted.
+    #[cfg_attr(
+        not(feature = "i-implement-a-third-party-backend-and-opt-into-breaking-changes"),
+        doc(hidden)
+    )] // This is used by raw SQL fragments
+    #[cfg_attr(
+        doc_cfg,
+        doc(cfg(feature = "i-implement-a-third-party-backend-and-opt-into-breaking-changes"))
+    )]
+    pub fn push_bind_param_value_only<T, U>(&mut self, bind: &'b U) -> QueryResult<()>
```

Comment:
No need to to this kind of complicated setup here. We have an proc_macro attribute for that which takes care of all the things.

```suggestion
    #[diesel_derives::__diesel_public_if(
        feature = "i-implement-a-third-party-backend-and-opt-into-breaking-changes"
    )]
    pub(crate) fn push_bind_param_value_only<T, U>(&mut self, bind: &'b U) -> QueryResult<()>
```